ads/alp/handler.js

```diff
@@ -226,7 +226,7 @@ export function getA2AAncestor(win) {
     return null;
   }
   const amp = origins[origins.length - 2];
-  if (!ampOrigins.hasOwnProperty(amp)) {
+  if (!ampOrigins[amp] && !ampOrigins.hasOwnProperty(amp)) {
```

Comment:
I felt a comment might be more confusing. Before this launches we will definitely have compiled integration tests.